lighthouse-core/gather/computed/dependency-graph/estimator/estimator.js

```diff
@@ -0,0 +1,284 @@
+/**
+ * @license Copyright 2017 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Node = require('../node');
+const TcpConnection = require('./tcp-connection');
+
+// see https://cs.chromium.org/search/?q=kDefaultMaxNumDelayableRequestsPerClient&sq=package:chromium&type=cs
+const DEFAULT_MAXIMUM_CONCURRENT_REQUESTS = 10;
+const DEFAULT_RESPONSE_TIME = 30;
+const DEFAULT_RTT = 150;
+const DEFAULT_THROUGHPUT = 1600 * 1024; // 1.6 Mbps
+
+function groupBy(items, keyFunc) {
+  const grouped = new Map();
+  items.forEach(item => {
+    const key = keyFunc(item);
+    const group = grouped.get(key) || [];
+    group.push(item);
+    grouped.set(key, group);
+  });
+
+  return grouped;
+}
+
+class Estimator {
+  /**
+   * @param {!Node} graph
+   * @param {{rtt: number, throughput: number, defaultResponseTime: number,
+   *    maximumConcurrentRequests: number}=} options
+   */
+  constructor(graph, options) {
+    this._graph = graph;
+    this._options = Object.assign(
+      {
+        rtt: DEFAULT_RTT,
+        throughput: DEFAULT_THROUGHPUT,
+        defaultResponseTime: DEFAULT_RESPONSE_TIME,
```

Comment:
post-discussion:  naming-wise this value is very unclear.   defaultServerResponseTime is still ambiguous wether it includes latency or not. (and 1-way or 2-way?)  

`defaultTTFB` is very clear as a definition, so I'd prefer using that as the argument. That means some adding/subtracting of latency here and there, but i think it's worth it so the terms are clear.

Reply:
renamed to `fallbackTTFB` and added latency subtraction 👍

Reply:
ok hang on, reopening the discussion :)

i'm gonna push back against this latency subtraction from TTFB deal a bit, I cherrypicked the change to my final fixes branch and it regressed accuracy by 0.3%

given we're significantly under reporting rtt in most cases i'm not sure we should be further reducing the download time estimates, how about as future exploration we try to do something fancy to identify/separate connection latency from server response time instead (I can imagine trying to determine the built-in connection latency by looking at all the requests, or comparing the TCP handshake time to the TTFB time etc but its all more researchy) and leave it be for now?

Reply:
as discsussed offline, i think it'd be great to explore identifing per-server RTT later. this comment sg. let's add it. :)

Reply:
update: saw the comment is added already. 👍